visualization/app/codeCharta/ui/centerMapButton/centerMapButton.component.ts

```diff
@@ -0,0 +1,38 @@
+import "./centerMapButton.component.scss"
+import { CameraChangeSubscriber, ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
+import { PerspectiveCamera } from "three"
+import { IRootScopeService } from "angular"
+
+export class CenterMapButtonController implements CameraChangeSubscriber {
+	private _viewModel: {
+		isMapCentered: boolean
+	} = {
+		isMapCentered: true
+	}
+
+	/* @ngInject */
+	constructor(private $rootScope: IRootScopeService, private threeOrbitControlsService: ThreeOrbitControlsService) {
+		ThreeOrbitControlsService.subscribe(this.$rootScope, this)
+	}
+
+	public onCameraChanged(camera: PerspectiveCamera, event: angular.IAngularEvent) {
+		this._viewModel.isMapCentered = this.isMapCentered(camera)
+	}
+
+	public fitMapToView() {
+		this.threeOrbitControlsService.autoFitTo()
+	}
+
+	private isMapCentered(camera: PerspectiveCamera) {
+		return camera.position
+			.clone()
+			.floor()
+			.equals(this.threeOrbitControlsService.defaultCameraPosition.clone().floor())
```

Comment:
From what I just tried: I didn't need to clone the `defaultCameraPosition`

Reply:
position should be cloned before using them for calculations